web/themes/joinup/joinup_theme.theme

```diff
@@ -940,3 +940,40 @@ function joinup_theme_form_comment_form_alter(&$form, FormStateInterface $form_s
   $form['field_body']['widget'][0]['value']['#theme'] = 'textarea__comment';
   $form['field_body']['widget'][0]['value']['#theme_wrappers'] = ['form_element__textarea__comment'];
 }
+
+/**
+ * Implements hook_preprocess_HOOK() for table--file-formatter-table.html.twig.
+ *
+ * Extracts the file extension and makes it available on the row level.
+ */
+function joinup_theme_preprocess_table__file_formatter_table(&$variables) {
+  if (!empty($variables['rows'])) {
+    foreach ($variables['rows'] as $index => $row) {
+      /** @var \Drupal\file\Entity\File $file */
+      $file = $variables['rows'][$index]['cells'][0]['content']['#file'];
+      $file_icon = pathinfo($file->getFilename(), PATHINFO_EXTENSION);
+
+      // List of available icons.
+      $available_icons = [
+        'doc',
```

Comment:
These are ok. Strangely though, the prototype (http://localhost:3000/section/2.8) didn't show the icons, so I had to manually check each one.